site/docs/hosting/vps.md

```diff
@@ -50,131 +151,610 @@ cd /etc/systemd/system
+npm install -g pm2
+```
+
+```sh [Yarn]
+yarn global add pm2
+```
+
+```sh [pnpm]
+pnpm add -g pm2
+```
+
+:::
+
+#### Creating an Application
+
+PM2 offers two ways to create an application:
+
+1. Use the command line interface.
+2. Use the [configuration file](https://pm2.keymetrics.io/docs/usage/application-declaration).
+
+The first method is convenient when getting to know PM2.
+However, during deployment, you should use the second method, which is what we did in our case.
+
+On our server, in the directory where the bot assembly is stored, there is a file `ecosystem.config.js` with the following contents:
+
+```js
+module.exports = {
+  apps: [{
+    name: "<app-name>",
+    script: "<start-command>",
+  }],
+};
+```
+
+where `<app-name>` can be any identifier, and `<start command>` must be the actual command to start the bot, for example, `node bot.js` or `npm run start`, etc.
+
+#### Managing the Application
+
+Below are the commands you can use to control the application.
 
 ```sh
-systemctl start <service_name>
+# If the `ecosystem.config.js` file is in the current directory,
+# you can specify nothing to start the application.
+# If the application is already running, this command will restart it.
+pm2 start
+
+# All of the following commands require you to specify the name of the application
+# or the `ecosystem.config.js` file.
+# To apply the action to all applications, specify `all`.
+
+# To restart the app
+pm2 restart <<app-name>>
+
+# To reload the app
+pm2 reload <<app-name>>
+
+# To stop the app
+pm2 stop <<app-name>>
+
+# To delete an app
+pm2 delete <<app-name>>
 ```
 
-> Replace `<service_name>` with the file name of the service.
-> Example: `systemctl start bot1`
+#### Saving the Application Operation
+
+If the server reboots, your bot will not resume working.
+For the bot to resume work, you need to prepare PM2 for this.
 
-#### Run on Boot
+On the server in the terminal, run the following command:
 
 ```sh
-systemctl enable <service_name>
+pm2 startup
 ```
 
-#### Check Logs
+You will be provided with a command that you must execute to make PM2 automatically start after the server reboots.
+
+Then run one more command:
 
 ```sh
-systemctl status <service_name>
+pm2 save
+```
+
+This command will save the list of current applications so that they can be launched after a server reboot.
+
+If you have created a new application and want to save it as well, simply run `pm2 save` again.
+
+## Running the Bot on Webhooks
+
+To run a bot on webhooks, you will need to use a web framework and **NOT** call `bot.start()`.
+
+Here is a sample code to run the bot on webhooks that should be added to the main bot file:
+
+::: code-group
+
+```ts [Node.js]
+import { webhookCallback } from "grammy";
+import { fastify } from "fastify";
+
+const server = fastify();
+
+server.post(`/${bot.token}`, webhookCallback(bot, "fastify"));
+
+server.listen();
+```
+
+```ts [Deno]
+import { serve } from "https://deno.land/std/http/server.ts";
```

Comment:
This part of std was discontinued. We need to switch to `Deno.serve`. However, given that this PQ has existed for such a long time already, we may want to fix this in a subsequent PQ. Another reason for doing this later is that we probably have to adjust several more places in the docs regarding this, so lets leave it for now.

Reply:
#881 has changed my mind, we should fix it here